src/main/java/org/jenkinsci/plugins/credentialsbinding/MultiBinding.java

```diff
@@ -138,4 +141,19 @@ public Unbinder getUnbinder() {
         return (BindingDescriptor<C>) super.getDescriptor();
     }
 
+    /**
+     * Utility method for turning a collection of secret strings into a {@link Secret}.
+     * @param secrets A collection of secret strings
+     * @return A {@link Secret} generated from that collection.
+     */
+    public static Secret getSecretForStrings(Collection<String> secrets) {
+        StringBuilder b = new StringBuilder();
+        for (String secret : secrets) {
+            if (b.length() > 0) {
+                b.append('|');
+            }
+            b.append(Pattern.quote(secret));
+        }
+        return Secret.fromString(b.toString());
```

Comment:
And only `BindingStep` needs it to be a `Secret`. `SecretBuildWrapper` just throws that away and recompiles the `Pattern`—a waste, and confusing.